python/lsst/obs/test/ingestCalibs.py

```diff
@@ -0,0 +1,74 @@
+# This file is part of obs_test.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""Custom task for ingesting obs_test calibration files.
+"""
+
+__all__ = ["TestCalibsParseTask"]
+
+import os
+from lsst.pipe.tasks.ingestCalibs import CalibsParseTask
+
+
+class TestCalibsParseTask(CalibsParseTask):
+    """Specialization of `CalibsParseTask` for files with minimal headers.
+    """
+    def getCalibType(self, filename):
+        """Return a a known calibration dataset type using the filename.
+
+        Parameters
+        ----------
+        filename : `str`
+            Input filename
+
+        Returns
+        -------
+        type : one of "bias", "dark", "flat", "fringe", "sky", "defect", "unknown"
+            The type of dataset.
+        """
+        basename = os.path.basename(filename)
+        obstype = "unknown"
+        if "flat" in basename:
+            obstype = "flat"
+        elif "bias" in basename:
+            obstype = "bias"
+        elif "defects" in basename:
+            obstype = "defect"
+        return obstype
+
+    def translate_date(self, md):
+        """Convert a full TIME-MID to a mere date.
+
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertySet`
+            FITS header metadata
+
+        Returns
+        -------
+        date : `str`
+            The date, or "unknown" if the value cannot be found or converted.
+        """
+        date = md.get("TIME-MID").strip()
+        c = date.find("T")
```

Comment:
Consider `return date.split("T")[0]`. Like your code it will return everything up to, but not including, the first `T`. If there is no T then it returns the whole `date`.

Using `split` in this way is a common Python idiom.